src/system/SystemLayerImplSelect.cpp

```diff
@@ -269,7 +277,10 @@ void LayerImplSelect::CancelTimer(TimerCompleteCallback onComplete, void * appSt
 #endif // CHIP_SYSTEM_CONFIG_USE_DISPATCH/LIBEV
 
     mTimerPool.Release(timer);
+#if !CHIP_SYSTEM_CONFIG_USE_LIBEV
+    // LIBEV implementation should never call Signal()
```

Comment:
Here (and in the log as well maybe) we should explain why signal should never be called. There seems to be a difference between "should never" (meaning it is an error ... in which case why is signal not a NOOP under libev) or "it is not useful".